src/common/module_builder.cc

```diff
@@ -192,9 +241,15 @@ void ModuleBuilder::createFlatbufferBinary(
   tt::runtime::Binary runtime_binary_handle(m_flatbuffer_binary);
   m_num_inputs = runtime_binary_handle.getProgramInputs(0).size();
   m_num_outputs = runtime_binary_handle.getProgramOutputs(0).size();
+
+  if (m_num_outputs != m_is_output_scalar.size()) {
+    DLOG_F(ERROR, "The number of return types does not match ");
```

Comment:
I would change the message to: `"Created flatbuffer binary contains different number of outputs ({%0}) than expected ({%1})", m_num_outputs , m_is_output_scalar.size()`. I am not sure how exactly formatting is done in loguru to insert these numbers, you can look it up.